jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -291,68 +285,69 @@ def search(self, error_on_fail=False):
                     sender=self)
             self.hub.broadcast(snackbar_message)
 
-        # coordinates found are converted to pixel coordinates
-        pixel_table = viewer.state.reference_data.coords.world_to_pixel(skycoord_table)
-        # coordinates are filtered out (using a mask) if outside the zoom range
-        pair_pixel_table = np.dstack((pixel_table[0], pixel_table[1]))
-        # ma.masked_outside removes the coordinates outside the zoom range
-        # ma.compress_rows removes any row that has a mask mark
-        filtered_table = ma.compress_rows(
-            ma.masked_outside(pair_pixel_table, [zoom_x_min, zoom_y_min], [zoom_x_max, zoom_y_max])
-            [0])
-        # coordinates are split into their respective x and y values
-        # then they are converted to sky coordinates
-        filtered_pair_pixel_table = np.array(np.hsplit(filtered_table, 2))
-        x_coordinates = np.squeeze(filtered_pair_pixel_table[0])
-        y_coordinates = np.squeeze(filtered_pair_pixel_table[1])
-
+        # Convert to pixel coordinates and filter results to be within viewer bounds
         if self.catalog_selected in ["SDSS", "Gaia"]:
-            # for single source convert table information to lists for zipping
-            if len(self.app._catalog_source_table) == 1 or self.max_sources == 1:
-                x_coordinates = [x_coordinates]
-                y_coordinates = [y_coordinates]
+            skycoords = SkyCoord(self.app._catalog_source_table['ra'],
+                                 self.app._catalog_source_table['dec'],
+                                 unit='deg')
+        elif self.catalog_selected in ["From File..."]:
+            skycoords = self.app._catalog_source_table['sky_centroid']
+
+        pixel_table = viewer.state.reference_data.coords.world_to_pixel(skycoords)
+        self.app._catalog_source_table['x_coord'] = pixel_table[0]
+        self.app._catalog_source_table['y_coord'] = pixel_table[1]
```

Comment:
I feel like this is going to be problematic when the reference data is the fake GWCS layer. But also looks like this is existing logic, so perhaps battle for another day.

Reply:
I thought the same, but it turns out we disallow changing the linking between WCS and pixel after a catalog search/markers are added to avoid that issue. Definitely something we could think about improving later.

Reply:
If it was already WCS linked when the catalog is searched, all the pixel values do correspond to that layer, so they're between 0 and 10.